playbooks/roles/ocp-sso/tasks/main.yaml

```diff
@@ -0,0 +1,232 @@
+      delay: 10
+
+    - name: Check if pods are running
+      shell: oc get deploy -n openshift-secondary-scheduler-operator --no-headers | wc -l
+      register: sso_pods
+      until: sso_pods.stdout|int == 2
+      retries: 5
+      delay: 10
+
+- name: Run e2e test cases
+  block:
+  - name: Install dependencies for running e2e tests
+    dnf:
+      name: make
+      state: present
+
+  - name: Include role for installation of Go lang
+    include_role:
+      name: golang-installation
+    vars:
+      go_tarball: "{{ sso_go_tarball }}"
+      golang_path: "/usr/local"
+
+  - name: Clone the opeshift-test-private repository
+    git:
+      repo: "https://{{ sso_github_username }}:{{ sso_github_token }}@github.com{{ sso_e2e_github_repo | urlsplit('path') }}"
+      dest: "{{ sso_directory }}/openshift-tests-private"
+      version: "{{ sso_e2e_github_branch }}"
+
+  - name: Run make target
+    make:
+      chdir: "{{ sso_directory }}/openshift-tests-private"
+      target: build
+
+  - name: Generate SSO Operator test cases list
+    shell: |
+      ./bin/extended-platform-tests run all --dry-run | grep "secondary-scheduler\|74405" > sso-e2e-test.txt
+    args:
+      chdir: "{{ sso_directory }}/openshift-tests-private"
+    ignore_errors: true
+
+  - name: Run Secondary Scheduler Operator e2e tests
+    shell: |
+      ./bin/extended-platform-tests run -f sso-e2e-test.txt -o sso-e2e-output.txt
+    args:
+      chdir: "{{ sso_directory }}/openshift-tests-private"
+    ignore_errors: true
+
+  environment: "{{ sso_e2e_env }}"
+  ignore_errors: true
+  when: sso_e2e
+
+- name: Cleanup SSO operator
+  block:
+   - name: Check if the namespace exists
+     shell: oc get ns openshift-secondary-scheduler-operator --no-headers --no-headers | wc -l
+     register: sso_namespace
+     changed_when: false
+
+   - name: Delete the namespace if it exists
+     shell: oc delete ns openshift-secondary-scheduler-operator
+     when: sso_namespace.stdout | int > 0
+     register: delete_namespace
+
+   - name: Check if the namespace is stuck in Terminating state
+     shell: kubectl get ns openshift-secondary-scheduler-operator -o json | grep '"phase":"Terminating"' | wc -l
+     register: terminating_namespace
+     changed_when: false
+     when: sso_namespace.stdout | int > 0
+
+   - name: Force delete namespace if stuck in Terminating state
+     shell: |
+       kubectl get namespace openshift-secondary-scheduler-operator -o json \
+         | tr -d "\n" \
+         | sed 's/"finalizers": \[[^]]\+\]/"finalizers": []/' \
+         | kubectl replace --raw /api/v1/namespaces/compliance/finalize -f -
+     when: terminating_namespace.stdout | int > 0
+     register: fix_terminating_namespace
+
+   - name: Wait for namespace to be deleted
+     shell: oc get ns openshift-secondary-scheduler-operator --no-headers | wc -l
+     register: verify_namespace
+     retries: 10
+     delay: 15
+     until: verify_namespace.stdout | int == 0
+
```

Comment:
For cleanup task can please refer this[ task](https://github.com/ocp-power-automation/ocp4-playbooks-extras/blob/main/playbooks/roles/ocp-coo/tasks/cleanup.yaml), and follow the same, because directly deleting the namespace is not right approach, delete all the resources which you have created during Deployment of SSO and then delete the namespace.

Reply:
Updated the clean up task.